### PR TITLE
Add multiarch image build support using github actions.

### LINKED ISSUE #264: [RFE] Enable multi-platform support for  Multus Dynamic Networks Controller CNI image.
**Is your feature request related to a problem? Please describe.**
The upstream image of the Multus Dynamic Networks Controller CNI (available at [this link](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pkgs/container/multus-dynamic-networks-controller)) currently supports only the `amd64` architecture. To extend its support, we should build a multi-platform image that includes support for the `s390x` and `arm64` architecture as well.

**Describe the solution you'd like**
Enable multi-platform support for building the image of Multus Dynamic Networks Controller CNI.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ashokpariya0**
These changes enable the building of the Multus Dynamic CNI image for multiple platforms (amd64, arm64, s390x) using github actions.

**What this PR does / why we need it**:
The upstream image of the Multus Dynamic Networks Controller CNI ([link](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pkgs/container/multus-dynamic-networks-controller)) currently supports only the amd64 architecture. These changes extend multi-platform support to include arm64 and s390x architectures, enhancing the compatibility of the Multus Dynamic Networks Controller image across a broader range of systems.

Cross-compilation is performed in the builder image using the TARGETOS and TARGETARCH build arguments, which specify the target OS and architecture. These arguments are automatically configured by the multi-architecture build process using docker buildx

More information: [[Docker Buildx Documentation](https://docs.docker.com/build/building/multi-platform/)](https://docs.docker.com/build/building/multi-platform/)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #264 

**Special notes for your reviewer** *(optional)*:

### CODE REVIEW

**Comment by maiqueb** [comment]
@ashokpariya0 would you please rebase ?

**Comment by ashokpariya0** [comment]
> @ashokpariya0 would you please rebase ?
>
Done.

**Review by maiqueb — APPROVED** [review]
Thank you.

From my limited perspective, this looks correct.

File: images/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+ENV TARGETARCH=${TARGETARCH:-amd64}
+
 RUN mkdir -p $GOPATH/src/github.com/maiqueb/multus-dynamic-networks-controller
 WORKDIR $GOPATH/src/github.com/maiqueb/multus-dynamic-networks-controller
 COPY . .
-RUN GOOS=linux CGO_ENABLED=0 go build -o /dynamic-networks-controller ./cmd/dynamic-networks-controller
+RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} CGO_ENABLED=0 go build -o /dynamic-networks-controller ./cmd/dynamic-networks-controller
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal
+FROM --platform=linux/${TARGETARCH} registry.access.redhat.com/ubi8/ubi-minimal AS final
```
[the remark this answers is not in the document]
Comment (ashokpariya0) [reply images/Dockerfile]: > nit: `AS final` isn't used as ref, imho better to drop it, in case someone theoretically adds another layer
>
Done.

**Review by oshoval — APPROVED** [review]
Thanks

**Review by maiqueb — APPROVED** [review]

**Comment by oshoval** [comment]
Thanks
can you please issue a release so it will build multi arch images?

**Comment by maiqueb** [comment]
> Thanks can you please issue a release so it will build multi arch images?

here you go, freshly baked: https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/releases/tag/v0.3.5

**Comment by ashokpariya0** [comment]
> https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/releases/tag/v0.3.5
>



> > Thanks can you please issue a release so it will build multi arch images?
> 
> here you go, freshly baked: https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/releases/tag/v0.3.5
>
Awesome! I verified the latest release image on s390x, and it worked perfectly! :)

**Pull request merged**